packages/elements/list/src/insertFragmentList.ts

```diff
@@ -49,35 +73,134 @@ export const insertFragmentList = (editor: PlateEditor) => {
+  };
+
+  const wrapNodeIntoListItem = (node: TDescendant): TDescendant => {
+    return node.type === listItemType
+      ? node
+      : {
+          type: listItemType,
+          children: [node],
+        };
+  };
+
+  /**
+   * Checks if the fragment only consists of a single LIC in which case it is considered the user's intention was to copy a text, not a list
+   */
+  const isSingleLic = (fragment: TDescendant[]) => {
+    if (fragment.length !== 1 || !isListRoot(fragment[0])) return false;
+
+    const root = fragment[0];
+    // start at root path
+    const queue: Path[] = [[]];
+    let foundLic = false;
+    while (queue.length) {
+      const path = queue.shift()!;
+      const node = Node.get(root, path);
+      if (!Element.isElement(node)) {
+        return false;
+      }
+
+      if (isList(node)) {
+        const children = Array.from(Node.children(root, path)).map(
+          ([, p]) => p
+        );
+        queue.push(...children);
+      } else if ((node as TDescendant).type !== listItemContentType) {
+        return false;
+      } else if (!foundLic) {
+        foundLic = true;
+      } else {
+        return false;
+      }
+    }
+
+    return true;
   };
 
   return (fragment: TDescendant[]) => {
     const liEntry = findNode(editor, {
-      match: { type: li.type },
+      match: { type: listItemType },
       mode: 'lowest',
     });
+    // not inserting into a list item, delegate to other plugins
+    if (!liEntry) {
+      return insertFragment(
+        isListRoot(fragment[0]) ? [{ text: '' }, ...fragment] : fragment
+      );
+    }
 
-    if (liEntry) {
-      const [, liPath] = liEntry;
+    // delete selection (if necessary) so that it can check if needs to insert into an empty block
+    Transforms.insertFragment(editor, [{ text: '' }]);
 
-      // FIXME: fork insertFragment for edge cases
-      return Transforms.insertNodes(
-        editor,
-        fragment.flatMap((node) => trimList(node)),
-        { at: Path.next(liPath), select: true }
+    const licEntry = findNode(editor, {
+      match: { type: listItemContentType },
+      mode: 'lowest',
+    });
+    if (!licEntry) {
+      return insertFragment(
+        isListRoot(fragment[0]) ? [{ text: '' }, ...fragment] : fragment
       );
     }
 
-    const filtered: TDescendant[] = isListRoot(fragment[0])
-      ? [{ text: '' }, ...fragment]
-      : fragment;
+    const [, liPath] = liEntry!;
+    const [licNode] = licEntry!;
+    const isEmptyNode = isCollapsed(editor.selection) && !Node.string(licNode);
+    const [first, ...rest] = fragment
+      .flatMap(trimList)
+      .map(wrapNodeIntoListItem);
+
+    let nodes;
+    let firstText;
+    if (isListRoot(fragment[0])) {
+      if (isSingleLic(fragment)) {
+        firstText = first;
+        nodes = rest;
+      } else if (isEmptyNode) {
+        // FIXME: is there a more direct way to set this?
+        const li = Node.get(editor, liPath) as Element;
```

Comment:
This logic for pasting inside an empty list item is very convoluted, and as I noticed it is not working properly after the update to the latest Slate and Plate version. I'll have to revisit that anyway, so if you have any suggestions what would be a better way to address this case let me know please as I am not very familiar with Slate's API.

One particular case I try to address here is something like this, let's say we want to copy a simple list such as this one (a LI `z` with a UL consisting of 2 other LIs `y` and `w`):

```
- z
  - y
  - w
- v
```

This will be pasted into a list such as this:

```
- a
  - b
  - <paste here>
    - d
    - e
```

I'm expecting the result of this operation will look like this:

```
- a
  - b
  - z
    - d
    - e
    - y
    - w
  - v
```

So, the first node will replace the empty LI and, in addition to that, it should keep the existing children of the empty LI and add the children of the newly pasted fragment. The rest of the LIs will just be inserted as siblings of the (previously) empty LI, just as it is the case now.